README.md

```diff
@@ -1 +1,69 @@
+- all arguments 
+- environment variables
+- standard input/output/error
+
+That allows the plugin to operate as if it were a native part of the Atlas CLI.
+
+## Developing a Plugin
+
+Plugins can be developed with any programming language, however, in this example plugin we are using Go. A plugin directory can contain any number of files, but it strictly requires two: an executable binary file and a `manifest.yml` file.
+
+### Manifest File
+The manifest file (`manifest.yml`) is an essential part of an Atlas CLI plugin and needs to live in the root folder of the release asset. Here is what the manifest file for this example plugin looks like:
+
+```yaml
+name: atlas-cli-plugin-example
+description: this is an example plugin
+version: 2.0.1
+github:
+    owner: mongodb
+    name: atlas-cli-plugin-example
+binary: binary
+commands: 
+    example: 
+        description: Root command of the atlas cli plugin example
+```
+
+| Field | Description | Optional |
+| - | - | - |
+| `name` | The name of the plugin | No
+| `description` | A brief description of the plugin. | No
+| `version` | The [semantic versioning](https://semver.org/) of the plugin.  | No
+| `github.owner` | The GitHub repository owner | Yes
+| `github.name`| The GitHub repository name | Yes
+| `binary`| The name of the plugin's binary file | No
+| `commands.<command_name>` | An object where the key is the command name and the value is a description of the command | No (a plugin needs at least one command)
+| `commands.<command_name>.description` | A description of the command | No
+
+> **Important:** 
+> - The `version` field in the manifest file is crucial for the Atlas CLI to identify the installed plugin version and manage updates. Ensure that this field is always aligned with the plugin's release version to enable correct updates.
+> - If your plugin is not part of a Github repository you can omit the entire `github` section. However, if your plugin is hosted on GitHub and you want to enable update functionality, you must include this section.
+
+
+You may notice that there is no `manifest.yml` file in this repository, only a `manifest.template.yml`. This is because we use a [GitHub Action](https://github.com/mongodb/atlas-cli-plugin-example/blob/master/.github/workflows/release.yml) to automatically generate the `manifest.yml` file. Every time a new release is published, the template is used to create `manifest.yml` file and the release's version is automatically inserted into the `version` field. The [GoReleaser](https://goreleaser.com/) is used to create release assets that contain the plugin's binary and the `manifest.yml`.
```

Comment:
```suggestion
You may notice that there is no `manifest.yml` file in this repository, only a `manifest.template.yml`. This is because we use a [GitHub Action](https://github.com/mongodb/atlas-cli-plugin-example/blob/master/.github/workflows/release.yml) to automatically generate the `manifest.yml` file. Every time a new release is published, the template is used to create the required `manifest.yml` file, and the release's version is automatically inserted into the `version` field. The [GoReleaser](https://goreleaser.com/) is used to create release assets that contain the plugin's binary and the `manifest.yml`.
```